extensions/wikia/BannerNotifications/js/BannerNotifications.js

```diff
@@ -0,0 +1,251 @@
+ * BannerNotifications.show('Some error message', 'error', $('.myDiv'), 3000)
+ */
+define('BannerNotification', [
+	'jquery',
+	'wikia.window',
+	'wikia.onScroll',
+	'BannerNotifications.templates.mustache',
+	'wikia.mustache'
+], function ($, window, onScroll, templates, mustache) {
+	'use strict';
+
+	var defaultTimeout = 10000,
+		types = {
+			'notify': 'blue',
+			'confirm': 'green',
+			'error': 'red',
+			'warn': 'yellow'
+		},
+		classes = Object.keys(types).join(' '),
+		closeImageSource = window.stylepath + '/oasis/images/icon_close.png',
+		$pageContainer,
+		wikiaHeader,
+		headerHeight,
+		modal,
+		backendNotification;
+
+	/**
+	 * Constructs jQuery element with the notification
+	 * @param {String} content
+	 * @param {String} [type]
+	 * @returns {jQuery}
+	 */
+	function createMarkup(content, type) {
+		return $(
+			mustache.render(templates.BannerNotifications, {
+				imageSource: closeImageSource,
+				content: content
+			})
+		).addClass(type).hide();
+	}
+
+	/**
+	 * Adds given markup to DOM
+	 * @param {jQuery} $element
+	 * @param {jQuery} $parentElement
+	 */
+	function addToDOM($element, $parentElement) {
+		// allow notification wrapper element to be passed by extension
+		var $parent = $parentElement || isModalShown() ? modal : $pageContainer,
+			$bannerNotificationsWrapper = $parent.find('.banner-notifications-wrapper');
+		if (!$bannerNotificationsWrapper.length) {
+			$bannerNotificationsWrapper = $('<div></div>').addClass('banner-notifications-wrapper');
+			$parent.prepend($bannerNotificationsWrapper);
+		}
+		$bannerNotificationsWrapper.prepend($element);
+
+		$element.fadeIn('slow');
+	}
+
+	function BannerNotification(content, type, $parent, timeout) {
+		if (content instanceof jQuery && content.hasClass('banner-notification')) {
+			//create a notification object from already existing markup
+			this.content = content.find('.msg').html();
+			this.$element = content;
+			this.$parent = content.parent();
+			this.hidden = false;
+		} else {
+			this.content = content;
+			this.$element = null;
+			this.$parent = $parent;
+			this.hidden = true;
+			this.type = type;
+			this.timeout = timeout;
+		}
+	}
+
+	BannerNotification.prototype.show = function () {
+		var self;
+		if (!this.hidden) {
+			return this;
+		}
+		if (!this.$element) {
+			this.$element = createMarkup(this.content, this.type);
+		}
+		this.setType(this.type);
+
+		setUpClose(this);
+
+		// Modal notifications have no close button so set a timeout
+		if (isModalShown() && typeof this.timeout !== 'number') {
+			this.timeout = defaultTimeout;
+		}
+
+		addToDOM(this.$element, this.$parent);
+
+		this.hidden = false;
+
+		// Close notification after specified amount of time
+		if (typeof this.timeout === 'number') {
+			self = this;
+			setTimeout(function () {
+				self.hide();
+			}, this.timeout);
+		}
+		return this;
+	};
+
+	BannerNotification.prototype.hide = function (callback) {
```

Comment:
Let's do so, it's not used anywhere (at least by my investigation, which could have ommited something) :)